### ISSUE TITLE
Prompt for user input before database password is overwritten

### ISSUE THREAD

**Issue by mneil** [stelligent/mu#344]
Looking at this line in the database upsert file https://github.com/stelligent/mu/blob/develop/workflows/database_upsert.go#L108 if there is an error with the ssm call to get a parameter (maybe it's intermittent network issue, maybe ssm is down in the region, or maybe some other unforseen random thing) mu plows ahead and sets the param with a new value.

Should we prompt the user for input before making this decision?

**Comment by cplee** [comment]
Good catch.  Ideally, we'd be able to inspect the error.  The only error that is safe to ignore is one about the parameter not being found:

```An error occurred (ParameterNotFound) when calling the GetParameter operation```

**Comment by mneil** [comment]
I started work on it. The last piece is the correct error handling. In #343 I'm removing the call to GetParam and instead using ParamExists to check if the parameter is available (so we never have the password except on creation) which will be easier to check for errors.

The database_upsert_test is also failing because of a missing password for kms (I believe). In that case the prompt is causing the test to fail. I mocked the cli input for testing but stopped short of requiring an instance of the prompt object for mocking the db upsert call. I'll probably end up having to add the cli object to another stuct as a property to mock it in tests.

Either way I'm going to hold on this until #343 is done
